src/util/fileinfo.h

```diff
@@ -0,0 +1,304 @@
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QUrl>
+#include <QtDebug>
+
+#include "util/assert.h"
+
+namespace mixxx {
+
+/// Helper functions and extensions for QFileInfo using Mixxx terminology.
+namespace fileinfo {
+
+/// Only intended for internal usage.
+inline const QFileInfo& assertNotRelative(const QFileInfo& fileInfo) {
+    // Detect potential, unintended side-effects by the outer context,
+    // i.e. when accessing the transient working directory to resolve
+    // of relative paths. The path should either be empty or absolute,
+    // i.e. not relative.
+    DEBUG_ASSERT(!fileInfo.isRelative());
+    return fileInfo;
+}
+
+/// Returns the permanent location of a file.
+inline QString location(const QFileInfo& fileInfo) {
+    return assertNotRelative(fileInfo).absoluteFilePath();
```

Comment:
Why the assertion? This is not clear from the name. I think the assert locic belongs to the calling context. The same applies to the functions below.

But maybe I have not yet understand the issue you try to fix.

Reply:
Our "location" is supposed to be valid without additional context, hence cannot be a relative path.

Reply:
OK.